### PR TITLE
feat: support throw/throwException/throwError

### PR DESCRIPTION
**Pull request by popomore**

### CODE REVIEW

**Comment by popomore** [comment]
Ref https://github.com/twada/chai-to-assert/pull/2

File: package.json [hunk 1]
```diff
@@ -37,7 +37,8 @@
   "scripts": {
     "fmt": "semistandard --fix",
     "lint": "semistandard --verbose | snazzy",
-    "test": "npm run lint && mocha test"
+    "test": "npm run lint && mocha test",
+    "cov": "istanbul cover ./node_modules/.bin/_mocha test"
```
Comment (popomore) [inline package.json]: I add a `cov` command, do you want to run coverage in travis-ci? we use https://codecov.io, and it's good.

Comment (twada) [reply package.json]: Taking coverage is very good practice, but not related to this PR. Would you remove this line from this PR?

FYI: I'm using [Coveralls](https://coveralls.io/)

File: test/test.js [hunk 2]
```diff
@@ -172,3 +172,54 @@ describe('.within(start,finish)', function () {
     after: 'assert(two < five || ten < two)'
   });
 });
+
+describe('.throw([errorLike], [errMsgMatcher])', function () {
```
Comment (twada) [inline test/test.js]: signature of expect.js's `throws` is different from chai's

- chai: `.throw([errorLike], [errMsgMatcher])`
- expect.js: `.throw([Function|RegExp])`

Therefore chai's testcase does not fit. You should modify them for expect.js (That's why I created `expect-js-to-assert`)

```js
describe('.throw([Function|RegExp])', function () {
  testTransform({
    before: 'expect(func).to.throw()',
    after: 'assert.throws(func)'
  });
  testTransform({
    before: 'expect(func).to.not.throw()',
    after: 'assert.doesNotThrow(func)'
  });
  testTransform({
    before: 'expect(func).to.throw(/error message/)',
    after: 'assert.throws(func, /error message/)'
  });
  testTransform({
    before: 'expect(func).to.not.throw(/error message/)',
    after: 'assert.doesNotThrow(func, /error message/)'
  });
  testTransform({
    before: 'expect(func).to.throw(handler)',
    after: 'assert.throws(func, handler)'
  });
  testTransform({
    before: 'expect(func).to.not.throw(handler)',
    after: 'assert.throws(func, handler)'
  });
  testTransform({
    before: 'expect(func).to.throwError()',
    after: 'assert.throws(func)'
  });
  testTransform({
    before: 'expect(func).to.not.throwError()',
    after: 'assert.doesNotThrow(func)'
  });
  testTransform({
    before: 'expect(func).to.throwException()',
    after: 'assert.throws(func)'
  });
  testTransform({
    before: 'expect(func).to.not.throwException()',
    after: 'assert.doesNotThrow(func)'
  });
});
```

File: transform.js [hunk 3]
```diff
@@ -153,6 +154,29 @@ module.exports = function transform (file, api, options) {
             j.binaryExpression('<=', subject, path.value.arguments[1])
           ));
         }
+      case 'throw':
+      case 'throwError':
+      case 'throwException':
+        var errorMatcher;
+        if (path.value.arguments.length < 2) {
```
Comment (twada) [inline transform.js]: So, conditional related to `path.value.arguments.length` should be removed.

**Review by twada — COMMENTED** [review]
Thank you for your PR. Difference between chai and expect.js bites us.

File: test/test.js (hunk 2 above)
Comment (popomore) [reply test/test.js]: My mistake

**Comment by popomore** [comment]
Updated

**Comment by twada** [comment]
@popomore Now LGTM. Thanks!

**Comment by twada** [comment]
[expect-js-to-assert 0.2.0](https://github.com/twada/expect-js-to-assert/blob/master/CHANGELOG.md#020-2017-04-28) is out. Thanks!

**Pull request merged**